### PR TITLE
added NumUtils

### PR DESCRIPTION
**Pull request by nimo23**
Changes proposed in this pull request:

-:tada: **NumUtils.java** to provide common "financial/statistical/math"-formulas with `Num`.

- [x] added an entry with related ticket number(s) to the unreleased section of `CHANGES.md` 

In future, we can move common calculations (needed for indicators or criteria, calculations currently duplicated on multiple places)  to `NumUtils.java`. 

If you encounter duplicate formulas used within any indicator/cirterion/etc., then it may be worth to consider to include a general formula within NumUtils (DRY = "dont repeat yourself").

### CODE REVIEW

File: ta4j-core/src/main/java/org/ta4j/core/utils/NumUtils.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+     * <pre>
+     * (1 + percentGrowth/100)pow(days) * initialCapital
+     * </pre>
+     * 
+     * @param initialCapital the initial capital
+     * @param percentGrowth  the percent growth in percentage format (e.g. 5%)
+     * @param time           the time (e.g. 1 year)
+     * @return the compound interest
+     */
+    public static Num compoundInterest(Num initialCapital, Num percentGrowth, int time) {
```
Comment (team172011) [inline ta4j-core/src/main/java/org/ta4j/core/utils/NumUtils.java:45]: This is nothing that I would expect to be in a NumUtils class

Comment (nimo23) [reply ta4j-core/src/main/java/org/ta4j/core/utils/NumUtils.java:45]: The `NumUtils` class should provide common "**financial/statistical/math**"-formulas..this also includes typcial formulas which can be used in indicators and criterions (for example, typical math formulas, like formulas for average, variance, regression formulas, etc. ). And the `NumUtils` should avoid the need to include something like apache-math or the like. The `compoundInterest` is an example of a  typcial financial formula (https://www.investopedia.com/terms/c/compoundinterest.asp). The NumUtils should also avoid duplicated formulas in indicators, criteria, etc. 

>This is nothing that I would expect to be in a NumUtils class

What would you expect to have in the NumUtils?

**Comment by team172011** [comment]
as far as not needed in indicators or criterions providing common "financial/statistical/math"-formulas is out of scope for ta4j. @nimo23 if you not agree please open a ticket for discussion

**Comment by nimo23** [comment]
Ok, no problem. I also think that this it out of scope.

**Pull request closed without merging**